calico_node/ip_detection.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"regexp"
+	"strings"
+)
+
+func detectIPs(ignoreInterfaces []string) ([]string, error) {
```

Comment:
I think a better signature for this would be:

```
// detectIPs returns a list of IPv4 address and a list of IPv6 addresses assigned across
// all interfaces on the host which don't match any string in ignoreInterfaces, or an error.
func detectIPs(ignoreInterfaces []string) ([]net.IPNet, []net.IPNet error)
```

It's more flexible to pass around IPNet objects as opposed to strings, and it's useful to pass back both IPv4 and IPv6 addresses.